### ISSUE TITLE
Quest#1249 "Missing Diplomat" failure causes NPC Mikhail to lock up

### ISSUE THREAD

**Issue by Stump876** [azerothcore/azerothcore-wotlk#5780]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [x] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Upon failure of quest [1249,](https://wowgaming.altervista.org/aowow/?quest=1249) NPC 9432 Mikhail will no longer interact with any player until server reset. Even if the failed quest is abandoned. He is both a quest ending point for quest 1248, and quest giver for quest 1249.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

From my understanding, failed quests should be able to be abandoned and reaccepted once the event has reset. The NPC should still be able to have quests turned in, and offer the event quest once the event has been reset.

##### STEPS TO REPRODUCE THE PROBLEM:

1. .quest add 1248
2. .go creature 9432
3. Turn in and accept the chain quest (1249)
4. Fail it miserably
5. Abandon the quest and try to reaccept 
6. Repeat steps 1 and 2 on another character and attempt step 3

##### EXTRA NOTES:

Original report from Chromiecraft 
[https://github.com/chromiecraft/chromiecraft/issues/599](https://github.com/chromiecraft/chromiecraft/issues/599)

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/6947789622de45479d2311fd5d1635af3a957b5d

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by anguaive** [comment]
This quest was fixed for a brief period by https://github.com/azerothcore/azerothcore-wotlk/pull/5070, but the ObjectGUID port (https://github.com/azerothcore/azerothcore-wotlk/pull/4885) broke the hashmap targeting that the script relies on. All scripts that use hashmap targeting are broken (there aren't that many though). I tested this on commit f4c2264 (contains the GUID port), and 91081f4 (the one before that). 

```diff
@@ -3632,7 +3629,7 @@ ObjectList* SmartScript::GetTargets(SmartScriptHolder const& e, Unit* invoker /*
                 // xinef: my addition
                 if (e.target.unitGUID.getFromHashMap)
                 {
-                    if ((target = ObjectAccessor::GetCreature(scriptTrigger ? *scriptTrigger : *GetBaseObject(), MAKE_NEW_GUID(e.target.unitGUID.dbGuid, e.target.unitGUID.entry, HIGHGUID_UNIT))))
+                    if ((target = ObjectAccessor::GetCreature(scriptTrigger ? *scriptTrigger : *GetBaseObject(), ObjectGuid::Create<HighGuid::Unit>(e.target.unitGUID.entry, e.target.unitGUID.dbGuid))))
                         l->push_back(target);
                 }
                 else
```



The quest fails when Tapoke "Slim" Jahn reaches his final waypoint and

**Comment by anguaive** [comment]
I was unable to figure out what's wrong, would appreciate someone more skilled taking a look

**Comment by FrancescoBorzi** [comment]
> his quest was fixed for a brief period by #5070, but the ObjectGUID port (#4885) broke the hashmap targeting that the script relies on. All scripts that use hashmap targeting are broken (there aren't that many though). I tested this on commit [f4c2264](https://github.com/azerothcore/azerothcore-wotlk/commit/f4c226423dc0811efccf8135b46bc7ff46aab940) (contains the GUID port), and [91081f4](https://github.com/azerothcore/azerothcore-wotlk/commit/91081f4ad8fed98d7bdc9c0ae13d05c86032df9f) (the one before that).

excellent investigation @anguaive , thanks..
@UltraNix can you please have a look at this?
